File: LeetCode/198. 打家劫舍.ts
var rob = function (nums: number[]): number {
  if (nums.length <= 2) {
    if (nums.length === 0) return 0;
    return nums.length === 1 ? nums[0] : Math.max(nums[0], nums[1]);
  }
  let dp = Array(nums.length);
  dp[0] = nums[0];
  dp[1] = Math.max(nums[1], nums[0]);
  for (let i = 2; i < nums.length; i++) {
    dp[i] = Math.max(dp[i - 2] + nums[i], dp[i - 1]);
  }
  return dp[nums.length - 1];
};

File: LeetCode/198. 打家劫舍.rs
use std::cmp::max;
impl Solution {
    pub fn rob(nums: Vec<i32>) -> i32 {
        if nums.len()<=2{
            return nums.into_iter().max().unwrap_or(0);
        }
        let mut dp=vec![0;nums.len()];
        dp[0]=nums[0];
        dp[1]=max(nums[0],nums[1]);
        for i in 2..dp.len(){
            dp[i]=max(dp[i-1],dp[i-2]+nums[i]);
        }
        dp[dp.len()-1]
    }
}